router/batchrouter/handle_lifecycle.go

```diff
@@ -238,6 +247,16 @@ func (brt *Handle) Shutdown() {
 	_ = brt.backgroundWait()
 }
 
+func (brt *Handle) refreshDestination(destination backendconfig.DestinationT) {
+	var err error
+	if slices.Contains(asyncDestinations, destination.DestinationDefinition.Name) {
+		brt.asyncdestinationmanager, err = asyncdestinationmanager.NewManager(&destination, brt.backendConfig)
+		if err != nil {
+			brt.logger.Error("BRT: error occurred while creating new instance of %v. %v", destination.Name, err)
```

Comment:
We are panicking in [Setup](https://github.com/rudderlabs/rudder-server/pull/3371/files?show-viewed-files=true&file-filters%5B%5D=#diff-2fa84a69a3bfa3b97f23fd9d560c84f2e284dc8c0bdb4222dcc18937e086f687R57) and not here. Need to be consistent.

Reply:
@ItsSudip